src/mscorlib/corefx/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -13,12 +13,15 @@ public partial class CompareInfo
         [SecurityCritical]
         private readonly Interop.GlobalizationInterop.SafeSortHandle m_sortHandle;
 
+        private readonly bool m_isAsciiEqualityOrdinal;
+
         [SecuritySafeCritical]
         internal CompareInfo(CultureInfo culture)
         {
             m_name = culture.m_name;
             m_sortName = culture.SortName;
             m_sortHandle = Interop.GlobalizationInterop.GetSortHandle(System.Text.Encoding.UTF8.GetBytes(m_sortName));
+            m_isAsciiEqualityOrdinal = (m_sortName == "en-US" || m_sortName == "");
```

Comment:
We can probably do this for all en locales (there are others as well).  I will land this and think about ways to extend it (I suspect we could look at the rules for a collator and understand if it's safe to preform this optimization)